airbyte-webapp/src/config/types.ts

```diff
@@ -38,4 +62,4 @@ export type Provider<T> = () => T;
 
 export type ValueProvider<T> = Array<ProviderAsync<DeepPartial<T>>>;
 
-export type ConfigProvider<T extends Config = Config> = ProviderAsync<DeepPartial<T>>;
+export type ConfigProvider<T extends AirbyteWebappConfig = AirbyteWebappConfig> = ProviderAsync<DeepPartial<T>>;
```

Comment:
I think most of those types could still be removed or the Provider which seems to be used in the auth service, just inlined there instead (and most likely doesn't need a named type).

Reply:
Thanks, forgot that I wanted to clean up (remove) these unnecessary types!

Reply:
addressed here https://github.com/airbytehq/airbyte/pull/21456/commits/1550b3a594946cdb242ec8c438503c5ca1969ffe